lib/spack/spack/cmd/unload.py

```diff
@@ -4,20 +4,59 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import argparse
-from spack.cmd.common import print_module_placeholder_help
+import sys
 
-description = "remove package from environment using `module unload`"
-section = "modules"
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.environment as ev
+import spack.util.environment
+import spack.user_environment as uenv
+
+description = "remove package from the user environment variables`"
+section = "user environment"
 level = "short"
 
 
 def setup_parser(subparser):
     """Parser is only constructed so that this prints a nice help
        message with -h. """
+    shells = subparser.add_mutually_exclusive_group()
+    shells.add_argument(
+        '--sh', action='store_const', dest='shell', const='sh',
+        help="print sh commands to activate the environment")
+    shells.add_argument(
+        '--csh', action='store_const', dest='shell', const='csh',
+        help="print csh commands to activate the environment")
+
     subparser.add_argument(
-        'spec', nargs=argparse.REMAINDER,
+        'specs', nargs=argparse.REMAINDER,
         help='spec of package to unload with modules')
 
 
 def unload(parser, args):
-    print_module_placeholder_help()
+    env = ev.get_env(args, 'unload')
+    specs = list(map(lambda spec: spack.cmd.disambiguate_spec(spec, env),
+                     spack.cmd.parse_specs(args.specs)))
```

Comment:
1. What happens if you load something, enter an environment, and then try to unload it?  Seems like it should still succeed.
2. Should loaded specs continue to be loaded if you activate an environment?  Or should env activate/deactivate unload/load everything in `SPACK_LOADED_HASHES`?  People are going to try to use these things together.
3. Shouldn't this just look in `SPACK_LOADED_HASHES` to disambiguate?  Why does it need to search everything?